### ISSUE TITLE
Local functions defined in cells persist across cells.

### ISSUE THREAD

**Issue by rsp34** [mathworks/jupyter-matlab-proxy#65]
Really like ["With MATLAB R2022b and later, you can define a local function at the end of the cell where you want to call it"](https://github.com/mathworks/jupyter-matlab-proxy).

Would be great if the functions could be called in other cells without re-writing the local function.

**Comment by prabhakk-mw** [comment]
Thank you for the `feature request` @rsp34 !
Our teams will consider this for a future release of the product.

From the outset, there may be challenges to implement this in a way that disambiguates functions with local cell scope from functions that have notebook scope.

Nonetheless a worthwhile feature request!

**Comment by prabhakk-mw** [comment]
Also requested by @christianschroder [here](https://blogs.mathworks.com/matlab/?p=1331#reply_2531749).

In-lining his thoughts below:

> BTW, one remark: it would be nice if, for larger notebooks, it were possible to declare functions in the standard manner (function out = myfun(in) ... end) in cells of their own, and have them be available for all cells subsequently executed.
Right now, as far as I can tell a function declared this way is only callable within the cell, so

```matlab
myfun(3, 4)
function c = myfun(a, b)
    c = a^2 + b^2;
end
```

> will work and output 25, but writing

```matlab
function c = myfun(a, b)
    c = a^2 + b^2;
end
```
> and then
```matlab
myfun(3, 4)
```
> in subsequent cells will not, instead giving you an error **(Unrecognized function or variable 'myfun'.)**
Anonymous functions and function handles work, of course - but I'm sure you can see why supporting proper, named functions would be useful, so I'd like to suggest this for the future.

**Comment by tylerlekang** [comment]
Yes, this would be very helpful. One way that I used python notebooks in grad school was to define a main function in a high-up cell that would analyze results for a scenario of parameter values, then plot the results in various ways. In the subsequent cells you could just create the scenarios are call that analysis and plotting function.

In Matlab notebooks, you can of course define a function in its own .m file and put that on the path, but then you lose access/ability to read that function code right as part of the notebook.

**Comment by Shushant2502** [comment]
Hello @rsp34,

As of version `v0.14.0`, the MATLAB Kernel now supports Magic commands, enabling you to use `%%lsmagic`, `%%help`, `%%time`, and `%%file` magics with the Kernel. The `%%file` magic can be used to define local functions directly within a cell. You can find a brief guide on this in the Magics [README](https://github.com/mathworks/jupyter-matlab-proxy/tree/main/src/jupyter_matlab_kernel/magics#create-new-functions-using-the-the-file-magic-command).

Thank you for using MATLAB Integration for Jupyter!

**Comment by dpo** [comment]
@Shushant2502 That is definitely an improvement, but I wonder if this could be done automatically for functions defined in cells. Coming from Jupyter with Python or Julia kernels, we're used to defining a function in a cell and calling it in another. I recognize that Matlab looks for a file with the same name as the function, but it feels like a major throwback.
